pkg/datastore/deploymentstore.go

```diff
@@ -97,9 +97,9 @@ var (
 )
 
 type DeploymentStore interface {
-	AddDeployment(ctx context.Context, d *model.Deployment) error
-	GetDeployment(ctx context.Context, id string) (*model.Deployment, error)
-	ListDeployments(ctx context.Context, opts ListOptions) ([]*model.Deployment, string, error)
+	Add(ctx context.Context, d *model.Deployment) error
+	Get(ctx context.Context, id string) (*model.Deployment, error)
+	List(ctx context.Context, opts ListOptions) ([]*model.Deployment, string, error)
 	UpdateDeployment(ctx context.Context, id string, updater func(*model.Deployment) error) error
 	UpdateDeploymentMetadata(ctx context.Context, id string, metadata map[string]string) error
 	UpdateDeploymentStageMetadata(ctx context.Context, deploymentID, stageID string, metadata map[string]string) error
```

Comment:
[nits] Is it more suitable that also remove "Deployment" from UpdateDeploymentXXX? but it's up to you.
```suggestion
	Update(ctx context.Context, id string, updater func(*model.Deployment) error) error
	UpdateMetadata(ctx context.Context, id string, metadata map[string]string) error
	UpdateStageMetadata(ctx context.Context, deploymentID, stageID string, metadata map[string]string) error
```

Reply:
Sure, it's a bit complicated to move all UpdateFunc, so I plan to refactor it after this PR be merged to reduce the diff 🙏